### DISCUSSION TITLE
Should time parameter (add_individual) of parent and offspring be set to 1 and 0?

### DISCUSSION THREAD

**Discussion by lakishadavid** [tskit-dev/tsinfer#551] [Q&A] [answered]
When using `add_individual(ploidy=1, metadata=None, population=None, location=None, time=0, flags=None)` on parent-offspring data, should parents' time be set to 1 and their offspring to 0? And with grandparents, should their time parameter be set to 2? Am I understanding this correctly?

**Comment by hyanwong** [comment, marked as answer]
No, the time in tsinfer is not measured in generations, sorry. 

In general, there is no standard meaning of time in a tree sequence - it depends on how it was created. For tree sequence created from tsinfer, the "time" ranges from 0 to 1, and simply measures the frequency of the focal derived allele. 

For tsinfer we assume that all input genomes are "samples" and not internal nodes (i.e. they are not the ancestors of other individuals in the tree sequence, although something that does indeed happen.

We haven't thought much about how to incorporate knowledge of the pedigree into the inference process. It should probably work just treating them as samples, all at time 0, but caveat emptor.

To turn the dates from tsinfer into something meaningful, you'll need to *date* the tree sequence, e.g. using [tsdate](https://github.com/tskit-dev/tsdate). Note that the dates for coalescence times are likely to be many orders of magnitude larger than single generations. E.g. the average coalescent time of two alleles from the same individual is of the order of thousands of generations (hundreds of thousands of years)

**Reply by hyanwong** [reply]
Hi @lakishadavid - would you like to mark this as "answered", if the explanation above is clear enough?
